packages/wrangler/src/pages/deploy.tsx

```diff
@@ -337,9 +337,42 @@ export const Handler = async (args: PagesDeployArgs) => {
 		project_name: projectName,
 	});
 
-	logger.log(
-		`✨ Deployment complete! Take a peek over at ${deploymentResponse.url}`
-	);
+	let latestDeploymentStage: DeploymentStage | undefined;
+
+	// can this ever become an infinite loop?
+	while (
+		latestDeploymentStage?.name !== "deploy" &&
+		latestDeploymentStage?.status !== "success" &&
+		latestDeploymentStage?.status !== "failure"
+	) {
+		try {
+			const deployment = await fetchResult<Deployment>(
+				`/accounts/${accountId}/pages/projects/${projectName}/deployments/${deploymentResponse.id}`
+			);
+			latestDeploymentStage = deployment.latest_stage;
+		} catch (err) {
+			logger.debug(
```

Comment:
- I'm assuming we don't want to exit here?
- Is `debug` log level OK?
- would also appreciate a check on the wording. I feel like in the crr state it might confuse folks and needs some resolution at the end?

Reply:
all lgtm!